### PR TITLE
feat(PostNewJob-ui):  Opening modal and everything(chapter 1)

### PR DESCRIPTION
**Pull request by JetoPistola**
Screen Shot 2020-03-31 at 19 06 23

Rest of comments in Teams. Will be happy to respond on any follow up questions you guys have 😃 

*** Update
This is chapter 1, looks as in the image above. Has fields, some logic of `avoid illegal input`, lower/upper select in department, datepicker, some tool tips, Assistant font, Themes override and such.

Chapter 2 will have:
1. Validations (Asterisk + Red/Animation + ErrorText)
2. 2 Swals/Dialogs (Success + error)
3. Storing in DB

Thanks in advance for feedback,

Daniel 😃

### CODE REVIEW

File: src/types/Departments.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+    getDepartmentFields = (): string[] => {
+        return Object.keys(DEPARTMENT_TO_DISPLAY);
+    }
+
+    getDepartmentFieldDisplay = (field: string): string => {
+        return DEPARTMENT_TO_DISPLAY[field];
+    }
+
+    getDepartmentFieldOptions = (department: DepartmentData, field: string): string[] => {
```
Comment (michael-hoffman-26) [inline src/types/Departments.ts]: Maybe add comment to make it more clear

File: src/types/Departments.ts [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        )();
+
+        return `יש לבחור ${this.getDepartmentFieldDisplay(higherField)} קודם`;
+    }
+
+    updateDepartment = (department: DepartmentData, field: string, value: string): DepartmentData => {
+        let updatedDepartment: DepartmentData = department;
+
+        switch (field) {
+            case 'unit': /* fall through */
```
Comment (michael-hoffman-26) [inline src/types/Departments.ts]: add comments

File: src/components/Application-Body/PostNewJob.tsx [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+                                    vertical: "bottom",
+                                    horizontal: "center"
+                                },
+                                transformOrigin: {
+                                    vertical: "top",
+                                    horizontal: "center"
+                                },
+                                getContentAnchorEl: null
+                            }}
+                            onChange={(event: any) =>
```
Comment (michael-hoffman-26) [inline src/components/Application-Body/PostNewJob.tsx]: without any

Comment (michael-hoffman-26) [reply src/components/Application-Body/PostNewJob.tsx]: again

File: src/types/Departments.ts (hunk 1 above)
Comment (JetoPistola) [reply src/types/Departments.ts]: Done 😄

File: src/types/Departments.ts (hunk 2 above)
Comment (JetoPistola) [reply src/types/Departments.ts]: Done 😄

File: src/components/Application-Body/PostNewJob.tsx (hunk 3 above)
Comment (JetoPistola) [reply src/components/Application-Body/PostNewJob.tsx]: Done 😄

**Pull request merged**